pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -538,6 +547,13 @@ var EBSVolumeFilter = VolumeFilter{
 		}
 		return "", false
 	},
+
+	IsMigrated: func(csiNode *storagev1beta1.CSINode) bool {
+		if !isCSIMigrationOn(csiNode, csilibplugins.AWSEBSInTreePluginName) {
+			return false
+		}
+		return hasVolumeLimits(csiNode, csilibplugins.AWSEBSDriverName)
```

Comment:
I needed to wrap `isCSIMigrationOn` into the volume plugin filter because each in-tree volume should be checked against a different CSI driver (if the plugin is migrated).

> Can we remove hasVolumeLimits? If migration is on, then we should always use CSI predicate and not fallback to in-tree predicate regardless of if driver is installed or not.

The KEP states that we should check if the driver is installed or not:

> As described in section above - old predicates will be modified to perform an additional check of CSINode object at very beginning of their code. If they detect that CSINode object contains limit of same underlying volume type or CSI migration has been enabled for the volume, the old predicate will return early(with success) and MaxCSIVolumeCountPred will be responsible for counting both CSI and in-tree volumes of same type.

Should we drop this requirement? CC @gnufied

Reply:
As discussed above - since we are not going to switch counting of volumes from old predicate to new predicate if CSI driver is installed but migration is off, we may drop this check from here.

Reply:
Dropped